### ISSUE TITLE
No Plane constructor that takes point and normal

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot#21113]
**Godot version:** 3.1 unofficial

**Issue description:**
The most common case for constructing a plane is, in my experience, when you want to have a plane that passes through a point. There is no way in the current API to do that in a trivial way, since the only similar plane constructor is the one that accepts a distance from the origin. It is trivial for axis aligned planes, but it starts to become more complicated once the planes are not aligned with any of the cardinal planes.

**Suggestion:**
Have a constructor that accepts a point and a normal.

**Comment by Noshyaar** [comment]
Would you mind testing if this code works?
```swift
static func plane(normal: Vector3, p: Vector3) -> Plane:
	var n = normal.normalized()
	return Plane(n, p.dot(n))
```

**Comment by Chaosus** [comment]
Yeah, please test, and I could PR it to engine(if its ok)

**Comment by QbieShay** [comment]
@Noshyaar it has a problem with the side that the plane is facing, you could end up with a plane that is symmetrical in respect to the origin depending on the orientation of the normal

**Comment by QbieShay** [comment]
@Chaosus I dug a little bit into the code and it looks like the constructor is already there. I am not 100% sure it works because it is the same solution that Noshyaar proposed which suffer from the problem i mentioned.

**Comment by Noshyaar** [comment]
Can you give me the parameters that fail?

**Comment by QbieShay** [comment]
@Noshyaar I don't understand why from a math point of view, but it actually works. I am now trying to expose the existing constructor to the engine.

**Comment by Noshyaar** [comment]
I imagined something like this. `dot` converts the distance in red to green.
![image](https://user-images.githubusercontent.com/26753932/44519435-3ae4b480-a6f7-11e8-8d8f-7c2424b8e089.png)

**Comment by QbieShay** [comment]
@Noshyaar yup. Might be that the Variant call need to have an unique signature?
@vnen

**Comment by vnen** [comment]
It should work if it's the only `(Vector3, Vector3)` constructor. Should be just a matter of adding:

```cpp
_VariantCall::add_constructor(_VariantCall::Plane_init4, Variant::PLANE, "point", Variant::VECTOR3, "normal", Variant::VECTOR3);
```

to `variant_call.cpp`.

**Comment by QbieShay** [comment]
@vnen Tried it, GDScript gives parsing error :(

**Comment by vnen** [comment]
I noticed it checks the number of arguments, so currently it's not possible to have different constructors with the same arity. It will need a change in Variant to enable this.

**Comment by QbieShay** [comment]
@vnen do you think we should address this issue? I can imagine this problem happening in other classes too

**Comment by Zylann** [comment]
Still no way to do this in GDScript, I occasionnally need it but have to write a wrapper each time.

**Comment by Calinou** [comment]
@vnen Could this be implemented in Godot 4.0 with the new method binding system?

**Comment by aaronfranke** [comment]
This feature request is on hold until #37842 is merged, since we don't want to add a constructor in GDScript only to swap it around later. #37842 currently has the unit tests failing, but it's a bug in the unit tests, not a bug in the PR's changes.

**Comment by Zireael07** [comment]
Bump. Just ran into this issue.... Any news on the constructor problem?

**Comment by vnen** [comment]
It might be possible in current master as I think the Variant binding supports it now, but GDScript doesn't really deal with multiple constructors with the same arity (in terms of typing), it just picks the first it finds. I think someone could try to add the constructor and test it if works without issue in GDScript.

**Comment by HolonProduction** [comment]
v4.4.beta1.official [d33da79d3] has a constructor that takes point and normal and it is perfectly usable from GDScript.

![Image](https://github.com/user-attachments/assets/aaa8c7f6-1604-4527-9a8d-18038a63d565)
